### PR TITLE
Move incremental state to WeakSets

### PR DESCRIPTION
**Pull request by jakebailey**
It turns out that the extra props we shove on nodes in incremental parsing are just "did we previously incrementally parse this, if so assert" and "did we make an edit that means we have to reparse", both of which are only ever set to true, so they don't even need to be WeakMaps, they're just write-only WeakSets.

### CODE REVIEW

**Comment by jakebailey** [comment]
@typescript-bot perf test this

Doubt this will do much of anything though, given we don't really have great perf tests for incremental parsing.

**Comment by DanielRosenwasser** [comment]
It's a bit of a shame that we don't have any long-running perf tests that make edits. The closest thing here is the crawler which does a few random edits.

File: src/compiler/parser.ts [hunk 1]
```diff
@@ -9844,6 +9843,26 @@ namespace Parser {
     }
 }
 
+const incrementalInfo = {
+    incrementallyParsed: new WeakSet<SourceFile>(),
+    intersectsChange: new WeakSet<Node | NodeArray<Node>>(),
+};
```
Comment (jakebailey) [inline src/compiler/parser.ts:9849]: Probably these two should just be top-level variables, not one object.

**Review by sandersn — APPROVED** [review]

**Comment by jakebailey** [comment]
@typescript-bot perf test this

**Pull request merged**